install_config/registry/deploy_registry_existing_clusters.adoc

```diff
@@ -91,7 +94,147 @@ Daemonsets ensure that when nodes are created, they contain copies of a
+3. The docker-registry service must exist before the docker-registry pod is created.
+
+**Service IP shall be preserved** because master caches the docker-registry's internal address.
+Unless one of the following applies, the service IP and port will be used to construct docker
+image pull-spec of the newly pushed images.
+
+- `OPENSHIFT_DEFAULT_REGISTRY` environment variable not containing
+  the service IP is passed to the master process 
+- xref:extended_registry_configuration.adoc#setting-the-registry-hostname[
+  `imagePolicyConfig:internalRegistryHostname`] not containing service IP is set in the master
+  configuration file
+
+It's important to keep the pull-spec's registry part of managed images and image stream
+repositories the same for all the images in the cluster because some tools (e.g. `oc adm prune
+images`) determine the address of the docker registry from the existing images unless directly
+provided.
+
+It's recommended to use the DNS registry name instead of the service address in any case. But even
+though used, preserving the registry's service IP will ensure consistency with whatever
+image-related resource already exists in the cluster which will make possible future migration
+easier.
+
+Before reading further, keep in mind also the following recommendations:
+
+1. Back up the existing configuration.
+2. It's usually safer and faster to modify the existing working deployment than starting from a
+scratch.
+3. When the above is satisfied, you can return back to the earlier working version with
+`oc rollout undo ...`
+
+[NOTE]
+====
+Unfortunately the `oc rollout` is currently limited just to the DeploymentConfig. Do the backup
+for the DaemonSet installation.
+====
+
+==== Determining registry's internal hostname
+
+The internal hostname is used for example to update imagestream's `status.dockerImageRepository`
+whenever it's created or updated. It's set on each new image's `dockerImageReference`. And is used
+in
+xref:../../admin_guide/image_policy.adoc#image-policy-configuring-the-image-policy-admission-plug-in[admission]
+and xref:../../admin_guide/image_policy.adoc#configuring-registries-allowed-for-import[registry
+whitelisting].
+
+To determine whether the master is configured to use the
+xref:extended_registry_configuration.adoc#setting-the-registry-hostname[DNS name] of IP address,
+execute the following in arbitrary namespace:
+
+----
+$ oc create imagestream tmpsvcip && \
+  oc get -o jsonpath=$'{.status.dockerImageRepository}\n' \
+    is/tmpsvcip | sed 's,/.*,,' && oc delete is/tmpsvcip >/dev/null
+----
+
+If the output is an `<IP>:<port>` hostname (e.g. `172.30.1.1:5000`), it probably matches your
+current service IP. The following command can be used to verify that:
+
+----
+$ oc get -n default -o jsonpath=$'{.spec.clusterIP}:{.spec.ports[0].port}\n' \
+  svc/docker-registry
+----
+
+If it's not the case and no custom IP is configured, the master processes need to be
```

Comment:
if what isn't the case?  if they don't match?  again what if they were using a hostname?

and what if a custom IP *is* being used?  (custom ip for what? the service ip?)